dotcom-rendering/src/lib/article-count.ts

```diff
@@ -0,0 +1,32 @@
+import { storage } from '@guardian/libs';
+import {
+	getWeeklyArticleHistory,
+	incrementWeeklyArticleCount,
+} from '@guardian/support-dotcom-components';
+import { hasOptedOutOfArticleCount } from '../web/lib/contributions';
+import { incrementDailyArticleCount } from '../web/lib/dailyArticleCount';
+
+// Log an article view using the Slot Machine client lib
+// We should monitor this function call to ensure it only happens within an
+// article pages when other pages are supported by DCR.
+export const getArticleCount = async (pageId: string, keywordIds: string) => {
```

Comment:
it would be good to declare the return type here - `Promise<WeeklyArticleLog[] | undefined>`

Reply:
Isn’t it `WeeklyArticleHistory`?

Reply:
True, `type WeeklyArticleHistory = WeeklyArticleLog[]`

`undefined` means no article count (perhaps because they've opted out), but it's not a failure.
I think the promise would only fail if CMP failed.